### PR TITLE
Create UOM-Conversions while importing article-csv

### PR DESCRIPTION
**Pull request by dm-ps**
refs: https://github.com/metasfresh/metasfresh/issues/18349

### CODE REVIEW

**Review by metas-ts — CHANGES_REQUESTED** [review]

File: backend/de.metas.business.rest-api-impl/src/main/java/de/metas/rest_api/v2/product/ProductRestService.java [hunk 1]
```diff
@@ -380,6 +391,72 @@ private void createOrUpdateProductTaxCategories(
 		}
 	}
 
+	private void createOrUpdateUOMConversions(
+			@Nullable final List<JsonRequestUOMConversionUpsert> uomConversions,
+			@NonNull final ProductId productId,
+			@NonNull final SyncAdvise effectiveSyncAdvise)
+	{
+		final UOMConversionsMap uomConversionsMap = uomConversionDAO.getProductConversionsEvenInactive(productId);
```
[the remark this answers is not in the document]
Comment (dm-ps) [reply backend/de.metas.business.rest-api-impl/src/main/java/de/metas/rest_api/v2/product/ProductRestService.java:399]: changed the index `c_uom_conversion_product` to `c_uom_conversion_product ON c_uom_conversion (c_uom_id, c_uom_to_id, m_product_id) WHERE IsActive = 'Y'`

File: backend/de.metas.business/src/main/java/de/metas/uom/impl/UOMConversionDAO.java [hunk 2]
```diff
@@ -23,16 +11,29 @@
...
 public class UOMConversionDAO implements IUOMConversionDAO
 {
 	private static final Logger logger = LogManager.getLogger(UOMConversionDAO.class);
 
-	private final CCache<ProductId, UOMConversionsMap> productConversionsCache = CCache.<ProductId, UOMConversionsMap> builder()
+	private final CCache<ProductId, UOMConversionsMap> productConversionsCache = CCache.<ProductId, UOMConversionsMap>builder()
+			.tableName(I_C_UOM_Conversion.Table_Name)
+			.build();
+
+	private final CCache<ProductId, UOMConversionsMap> productConversionsIncludingInactiveCache = CCache.<ProductId, UOMConversionsMap>builder()
```
[the remark this answers is not in the document]
Comment (dm-ps) [reply backend/de.metas.business/src/main/java/de/metas/uom/impl/UOMConversionDAO.java:36]: dropped

File: backend/de.metas.business.rest-api-impl/src/main/java/de/metas/rest_api/v2/product/ProductRestService.java [hunk 3]
```diff
@@ -380,6 +391,72 @@ private void createOrUpdateProductTaxCategories(
...
+			@NonNull final JsonRequestUOMConversionUpsert jsonRequestUOMConversionUpsert,
+			@NonNull final ProductId productId,
+			@NonNull final SyncAdvise effectiveSyncAdvise,
+			@NonNull final UOMConversionsMap uomConversionsMap)
+	{
+		validateJsonRequestUOMConversionUpsert(jsonRequestUOMConversionUpsert);
+		final UomId fromUomId = getUomId(X12DE355.ofCode(jsonRequestUOMConversionUpsert.getFromUomCode()));
+		final UomId toUomId = getUomId(X12DE355.ofCode(jsonRequestUOMConversionUpsert.getToUomCode()));
+
+		final Optional<UOMConversionRate> existingUomConversionRate = uomConversionsMap.getRateIfExists(fromUomId, toUomId);
```
[the remark this answers is not in the document]
Comment (dm-ps) [reply backend/de.metas.business.rest-api-impl/src/main/java/de/metas/rest_api/v2/product/ProductRestService.java]: introduced `de.metas.rest_api.v2.uomconversion.UomConversionRestService`

File: backend/de.metas.business.rest-api-impl/src/main/java/de/metas/rest_api/v2/product/ProductRestService.java [hunk 4]
```diff
@@ -380,6 +391,72 @@ private void createOrUpdateProductTaxCategories(
...
+																													 existingUomConversionRate.get());
+
+				uomConversionDAO.updateUOMConversion(updateUOMConversionRequest);
+			}
+		}
+		else if (effectiveSyncAdvise.isFailIfNotExists())
+		{
+			throw MissingResourceException.builder()
+					.resourceName("C_UOM_Conversion")
+					.resourceIdentifier("{ M_Product_ID:" + productId.getRepoId() + ", C_UOM_ID: " + fromUomId + ", C_UOM_To_ID: " + toUomId)
```
[the remark this answers is not in the document]
Comment (dm-ps) [reply backend/de.metas.business.rest-api-impl/src/main/java/de/metas/rest_api/v2/product/ProductRestService.java]: changed to detail

**Review by metas-ts — APPROVED** [review]

**Pull request merged**